care/users/api/viewsets/users.py

```diff
@@ -286,3 +286,17 @@ def pnconfig(self, request, *args, **kwargs):
                 setattr(user, field, request.data[field])
         user.save()
         return Response(status=status.HTTP_200_OK)
+
+class CheckUsernameAPIView(GenericAPIView):
+    """
+    Checks availability of username by getting as query, returns 200 if available, and 409 otherwise.
+    """
+
+    permission_classes = (AllowAny, )
+
+    def get(self, request):
+        username = request.query_params.get('username')
+        user = User.objects.filter(username=username) # since users are soft deleted, there might be inactive/deleted users that had the same username
```

Comment:
You can delete the comment, since deleted users are also users in our system since audit logs are tied to their username, so we should NOT allow the username to be reused in any case.